dascore/utils/patch.py

```diff
@@ -428,40 +458,90 @@ def _format_datetime64(dt):
     return f"DAS__{net}__{sta}__{tag}__{start}__{end}"
 
 
-def get_dim_value_from_kwargs(patch, kwargs):
+def get_dim_axis_value(
+    patch: PatchType,
+    *,
+    args: tuple = tuple(),
+    kwargs: dict = FrozenDict(),
+    arg_keys: tuple[str] = ("dim", "coord", "dims", "coords"),
+    allow_multiple: bool = False,
+    allow_extra: bool = False,
+) -> tuple[DimAxisValue, ...]:
     """
-    Assert that kwargs contain one value and it is a dimension of patch.
+    Get dimension nane, index, and values from args/kwargs for a patch.
+
+    This is helpful for implementing flexible fetching of dimension name,
+    corresponding patch axis, and function specific values from args and
+    kwargs as inputs.
 
-    Several patch functions allow passing values via kwargs which are dimension
-    specific. This function allows for some sane validation of such functions.
+    Parameters
+    ----------
+    patch
+        The patch which contains desired dimensions.
+    args
+        A tuple of possible dimension names.
+    kwargs
+        A dict of dimension_name: value
+    arg_keys
+        Keys in the dictionary that indicate
+    allow_multiple
+        If True, allow multiple dimensions to be selected.
+    allow_extra
+        If True, do not raise an error if extra args or kwargs are found.
+
+    Returns
+    -------
+    Returns a tuple of:
+        ((dim, axis, value), (dim, axis, value), ...)
+    To support retreiving multiple values from the same inputs. If dim name
+    is found in args, its corresponding values is `None`.
 
-    Return the name of the dimension, its axis position, and its value.
+    Examples
```

Comment:
Thanks for including these examples in the docs. Super helpful as this function is being used in a lot of other functions.